d2core/d2records/difficultylevels_loader.go

```diff
@@ -27,7 +28,14 @@ func difficultyLevelsLoader(r *RecordManager, d *d2txt.DataDictionary) error {
 			LifeStealDivisor:       d.Number("LifeStealDivisor"),
 			ManaStealDivisor:       d.Number("ManaStealDivisor"),
 		}
-		records[record.Name] = record
+		switch record.Name {
+		case "Normal":
+			records[d2enum.DifficultyNormal] = record
+		case "Nightmare":
+			records[d2enum.DifficultyNightmare] = record
+		case "Hell":
+			records[d2enum.DifficultyHell] = record
+		}
```

Comment:
Doesn't this couple the loader to expect explicitly the strings "Normal", "Nightmare" and "Hell" ? what happens when Johnny Modmaker comes along and adds an "Easy" difficulty?

Reply:
Yep. You're right. If we really want to allow any difficulty we can't use a difficulty enum, can we?

Reply:
For the meantime, i suppose it is okay, but i can see us moving all of the hardcoded data somewhere like an `OpenDiablo2.mpq` in the future.